### DISCUSSION TITLE
difference between pycaret and scikit-learn

### DISCUSSION THREAD

**Discussion by Friday131** [pycaret/pycaret#1273] [General]
I use same data、model、para(the scikit-learn's para are get from pycaret's model) to build model, and the same test data to predict, but the predict result of two model have a big difference, but I don't know why

**Comment by ngupta23** [comment]
`pycaret` does a lot of data preprocessing before the model is trained. Also, depends on at what stage you are predicting and how. If you predict before finalizing the model, you will predict on the test set which may be different between what you are using manually with `scikit-learn`. Also, the splits that you use to train the model may be different. In summary, there are a lot of reasons why the results can be different. That is not an issue. What is more important is that the results should not be bad. Are you seeing a deterioration in performance metrics compared to manually doing it in `scikit-learn`?
